dissect/target/plugins/os/unix/_os.py

```diff
@@ -301,37 +319,30 @@ def _parse_os_release(self, glob: str | None = None) -> dict[str, str]:
                                 continue
         return os_release
 
-    def _get_architecture(self, os: str = "unix", path: str = "/bin/ls") -> str | None:
-        arch_strings = {
-            0x00: "Unknown",
-            0x02: "SPARC",
-            0x03: "x86",
-            0x08: "MIPS",
-            0x14: "PowerPC",
-            0x16: "S390",
-            0x28: "ARM",
-            0x2A: "SuperH",
-            0x32: "IA-64",
-            0x3E: "x86_64",
-            0xB7: "AArch64",
-            0xF3: "RISC-V",
-        }
-
-        for fs in self.target.filesystems:
-            if fs.exists(path):
-                fh = fs.open(path)
-                fh.seek(4)
-                # ELF - e_ident[EI_CLASS]
-                bits = unpack("B", fh.read(1))[0]
-                fh.seek(18)
-                # ELF - e_machine
-                arch = unpack("H", fh.read(2))[0]
-                arch = arch_strings.get(arch)
-
-                if bits == 1:  # 32 bit system
-                    return f"{arch}_32-{os}"
-                else:
-                    return f"{arch}-{os}"
+    def _get_architecture(self, os: str = "unix", path: Path | str = "/bin/ls") -> str | None:
+        """Determine architecture by reading an ELF header of a binary on the target.
+
+        Resources:
+            - https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#ISA
+        """
+
+        if not isinstance(path, TargetPath):
+            for fs in [self.target.fs, *self.target.filesystems]:
+                if (path := fs.path(path)).exists():
+                    break
+
+        if not path.exists():
+            return
+
+        fh = path.open("rb")
+        fh.seek(4)  # ELF - e_ident[EI_CLASS]
+        bits = fh.read(1)[0]
+
+        fh.seek(18)  # ELF - e_machine
+        e_machine = int.from_bytes(fh.read(2), "little")
+        arch = ARCH_MAP.get(e_machine, "unknown")
+
+        return f"{arch}_32-{os}" if bits == 1 and not arch[-2:] == "32" else f"{arch}-{os}"
```

Comment:
```suggestion
        if not isinstance(path, TargetPath):
            path = self.target.fs.path(path)

        for fs in [self.target.fs, *self.target.filesystems]:
            if (path := fs.path(path)).exists():
                fh = path.open("rb")
                fh.seek(4)  # ELF - e_ident[EI_CLASS]
                bits = fh.read(1)[0]

                fh.seek(18)  # ELF - e_machine
                e_machine = int.from_bytes(fh.read(2), "little")
                arch = ARCH_MAP.get(e_machine, "unknown")

                return f"{arch}_32-{os}" if bits == 1 and not arch[-2:] == "32" else f"{arch}-{os}"
```

Reply:
This should do it. Also, thanks for adding the test!

Reply:
I prefer the original code, your suggestion is not really how `Path` objects are supposed to be used.

Reply:
Made a small edit in the suggestion. I dont see why doing the `.exists()` check twice is desirable. 
Only doing the path exists check in the loop seems fine to me and avoids the above. There can be instances where `Path` objects are passed to the check in the loop now, but that seems negligible.

Reply:
Fair point! I did not consider this 😄. After this 101 the original solution makes more sense. So consider this resolved.